### ISSUE TITLE
_PdfPreviewState.dispose failed to call super.dispose.

### ISSUE THREAD

**Issue by insinfo** [DavBfr/dart_pdf#931]
**Describe the bug**
This error appears when I click the back button on the screen where the PDF is

**To Reproduce**
Code snippet to reproduce the behavior:
```dart
import 'dart:typed_data';

import 'package:flutter/material.dart';

import 'package:http/http.dart' as http;
import 'package:printing/printing.dart';

class CustomPdfView extends StatefulWidget {
  final String? url;
  final Uint8List? pdfData;
  final String pdfFileName;
  final bool useActions;

  const CustomPdfView({
    Key? key,
    this.url,
    this.pdfData,
    this.pdfFileName = 'arquivo.pdf',
    this.useActions = true,
  }) : super(key: key);
  @override
  _CustomPdfViewState createState() => _CustomPdfViewState();
}

class _CustomPdfViewState extends State<CustomPdfView> {
  bool isFailure = false;

  Future<Uint8List> getData() async {
    print('CustomPdfPageView@getData pdfData');
    if (widget.url != null) {
      print('CustomPdfPageView@getData 1');
      var resp = await http.get(Uri.parse(widget.url!));
      if (resp.statusCode == 200) {
        print('CustomPdfPageView@getData resp.statusCode ${resp.statusCode}');
        return resp.bodyBytes;
      } else {
        throw Exception(
            'Erro ao obter os dados da URL, verifique a sua conexão');
      }
    } else if (widget.pdfData != null) {
      print('CustomPdfPageView@getData 2 ');
      return widget.pdfData!;
    }
    print('CustomPdfPageView@getData 3');
    throw Exception('É obrigatorio passar a URL ou os Bytes');
  }

  @override
  Widget build(BuildContext context) {
    return FutureBuilder(
      future: getData(),
      builder: (_, snapshot) {
        if (snapshot.hasError) {
          return Center(child: Text('Erro ao carregar o PDF'));
        }
        if (snapshot.hasData) {
          if (snapshot.data is Uint8List) {
            var data = snapshot.data as Uint8List;
            return Column(
              children: [
                Container(
                  width: double.infinity,
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      IconButton(
                          onPressed: () async {
                            await Printing.sharePdf(
                                bytes: data, filename: widget.pdfFileName);
                          },
                          icon: Icon(Icons.share)),
                      SizedBox(width: 20, height: 71),
                      IconButton(
                          onPressed: () async {
                            await Printing.layoutPdf(
                                onLayout: (_) => data,
                                name: widget.pdfFileName);
                          },
                          icon: Icon(Icons.print))
                    ],
                  ),
                ),
                AspectRatio(
                  aspectRatio: 1 / 1.4142,
                  child: PdfPreview(
                    canChangePageFormat: false,
                    canChangeOrientation: false,
                    canDebug: false,
                    actions: [],
                    useActions: false,
                    previewPageMargin: EdgeInsets.zero,
                    build: (format) {
                      return Future.value(data);
                    },
                  ),
                ),
              ],
            );
          } else {
            return Center(child: Text('Erro ao carregar o PDF'));
          }
        }
        return Center(child: CircularProgressIndicator());
      },
    );
  }
}


Observer(builder: (_) {
    if (controller.pdfContrachequeState ==
        LoadingState.loading) {
      return Center(
        child: CircularProgressIndicator(),
      );
    }
    if (controller.pdfContrachequeState == LoadingState.none) {
      return Container();
    }
    if (controller.pdfContrachequeState == LoadingState.error) {
      return Center(
        child: Text('Erro ao carregar o contracheque'),
      );
    }

    return Container(
      padding: EdgeInsets.all(16),
      child: CustomPdfView(
          pdfData: controller.pdfContracheque,
          pdfFileName: 'contracheque.pdf'),
    );
  })
```

**Screenshots**
![image](https://user-images.githubusercontent.com/12227024/152027573-cfb1cbc1-0702-4f1c-8fa9-54b96afa648f.png)

**Flutter Doctor**

```console
PS C:\MyDartProjects\riodasostras\riodasostrasapp_mob> flutter doctor -v
[√] Flutter (Channel stable, 2.8.1, on Microsoft Windows [versÃ£o 10.0.22000.434], locale pt-BR)
    • Flutter version 2.8.1 at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (7 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\Users\isaque\AppData\Local\Android\sdk
    • Platform android-32, build-tools 32.0.0
    • Java binary at: C:\Program Files\Android\Android Studio1\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31702.278
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.63.2)
    • VS Code at C:\Users\isaque\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.32.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [versÃ£o 10.0.22000.434]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 97.0.4692.99
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 97.0.1072.69

• No issues found!
PS C:\MyDartProjects\riodasostras\riodasostrasapp_mob> 
```

**Desktop (please complete the following information):**
 - [ ] iOS
 - [ ] Android
 - [ ] Browser
 - [x] Windows
 - [ ] Linux

```console
Exception has occurred.
FlutterError (_PdfPreviewState.dispose failed to call super.dispose.
dispose() implementations must always call their superclass dispose() method, to ensure that all the resources used by the widget are fully released.)
```

**Comment by deepak786** [comment]
this is already resolved in master branch https://github.com/DavBfr/dart_pdf/issues/929

**Comment by insinfo** [comment]
When will a corrected version be released? how can i put the direct version from gihub in my project?

**Comment by deepak786** [comment]
It already released 5.7.2

**Comment by DavBfr** [comment]
Don't hesitate to [Buy Me A Coffee](https://www.buymeacoffee.com/JORBmbw9h).
